File: medium/cpp/1968.cpp
// https://leetcode.com/problems/array-with-elements-not-equal-to-average-of-neighbors/

class Solution
{
public:
    vector<int> rearrangeArray(vector<int>& nums)
    {
        const auto n = std::size(nums);
        
        std::sort(std::begin(nums), std::end(nums));
        
        std::size_t pos = 0;
        std::vector<int> ans(n);
        
        for (std::size_t i = 0; i < n; i += 2)
        {
            ans[i] = nums[pos++];
        }
        
        for (std::size_t i = 1; i < n; i += 2)
        {
            ans[i] = nums[pos++];
        }
        
        return ans;
    }
};

File: medium/python/1968.py
# https://leetcode.com/problems/array-with-elements-not-equal-to-average-of-neighbors/

class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        nums.sort()
        
        n = len(nums)
        
        ans = [0] * n
        pos = 0
        
        for i in range(0, n, 2):
            ans[i] = nums[pos]
            pos += 1
            
        for i in range(1, n, 2):
            ans[i] = nums[pos]
            pos += 1
            
        return ans
        